code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+
+
+/obj/item/slimecross/warping/orange
+	desc = "Creates a rune "
+	colour = "orange"
+	runepath = /obj/effect/warped_rune/orangespace
+	effect_desc = "Creates a rune burning with bluespace fire, anyone walking into the rune will ignite and teleport randomly as long as they are on fire"
+
+/obj/effect/warped_rune/orangespace
+	name = "orange space rune"
+	desc = "When all is reduced to ash, it shall be reborn from the depth of bluespace"
+	icon_state = "bluespace_fire"
+	max_cooldown = 50
+
+
+
+/obj/effect/warped_rune/orangespace/Initialize()
+	.=..()
+	RegisterSignal(T,COMSIG_ATOM_ENTERED,.proc/teleport_fire) //TODO : figure out how to use the shortened version of this
+	START_PROCESSING(SSprocessing, src)
+
+/obj/effect/warped_rune/orangespace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		teleport_fire() //triggers it every 5 seconds anyway
+
+
+
+
+///teleport people and put them on fire if they run into the rune.
+/obj/effect/warped_rune/orangespace/proc/teleport_fire()
+	if(!locate(/obj/effect/hotspot) in T) //doesn't teleport item but put them on fire anyway for good measure.
+		new /obj/effect/hotspot(T)
+
+	for(var/mob/living/F in T)
+		do_teleport(F, T, 3, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)//only set the teleport to one for now
+		F.adjust_fire_stacks(10)
+		F.IgniteMob()
+
+/*
+		for(F.fire_stacks > 0 && !istype(F,/mob/living/simple_animal)) //Keeps teleporting the guy as long as he is on fire (doesn't include simplemobs) WIP
+			do_teleport(F, get_turf(F), 5, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)
+ */
+
+
+
+
+
+
+
+
+
+
+/*The purple warp rune makes suture and ointment if you put cloth or plastic on it. */
+
+
+
+
+/obj/item/slimecross/warping/purple
+	colour = "purple"
+	runepath = /obj/effect/warped_rune/purplespace
+	effect_desc = "creates a rune that transforms plastic into suture and cloth into regenerative mesh"
+
+/obj/effect/warped_rune/purplespace
+	desc = "When all that was left were plastic walls and the clothes on their back, they knew what they had to do."
+	icon_state = "purplespace"
+	var/obj/item/stack/medical/suture/S          // S stands for suture because it just make sense
+	var/obj/item/stack/medical/mesh/M            //M is for mesh and you
+	max_cooldown = 30
+
+/obj/effect/warped_rune/purplespace/Initialize()
+	.=..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/purplespace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		transmute_heal()
+
+
+///""transforms"" cloth and plastic into suture and regenerative mesh
+/obj/effect/warped_rune/purplespace/proc/transmute_heal() //transmute plastic and cloth into suture and regenerative mesh
+	for(var/obj/item/stack/sheet/plastic/P in T)  //transmute Plastic into regenerative mesh
```

Comment:
you might wanna check that there's at least 1 sheet in these plastic and cloth stacks? idk, it's a pretty unlikely edge case scenario, but you never know when some bug somewhere is gonna let you make a cloth or plastic stack with 0 sheets in it.

you also might want to make the ratio of plastic or cloth to mesh or sutures be 2:1 instead of 1:1, since this rune is a bit TOO efficient right now, IMO (and meshes and sutures only stack up to 25 "sheets" per stack anyway, so it works out).

still, I really like the idea of this rune, and I think that this could be something that people might actually specifically ask xenobiologists to make for them, sort of like how roboticists and borgs specifically request yellow slime core cells.